src/lib/DataSourceDefinition/Base.ts

```diff
@@ -7,20 +7,37 @@ export default abstract class Base {
   static get label(): string {
     throw new Error("Not Implemented");
   }
+  static get configSchema(): ConfigSchemaType {
+    throw new Error("Not Implemented");
+  }
 
   constructor(config) {
     this.config = config;
   }
 
-  abstract execute(query);
+  abstract execute(query: string): Promise<any>;
 
-  abstract cancel();
+  // @todo Set return type as Promise<void> ?
+  abstract cancel(): void | Promise<void>;
```

Comment:
BigQuery, MySQL and PostgreSQL uses Promise, but other not use.
I suggest to every datasource returns `Promise<void>`.